packages/openapi-typescript/src/transform/components-object.ts

```diff
@@ -51,14 +53,31 @@ export default function transformComponentsObject(
     const items: ts.TypeElement[] = [];
     if (componentsObject[key]) {
       for (const [name, item] of getEntries(componentsObject[key], ctx)) {
-        const subType = transformers[key](item, {
+        let subType = transformers[key](item, {
```

Comment:
Yup I think this will work in all cases I can think of. Just handling keys on object types should be the main usecase here (we don’t ever want to have question tokens, for example, directly on the `components["schemas"]` object as that doesn’t make sense)